src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Key.java

```diff
@@ -385,8 +388,10 @@ static PrivateKey privateKey(Session session, long keyID, String algorithm,
             new CK_ATTRIBUTE(CKA_SENSITIVE),
             new CK_ATTRIBUTE(CKA_EXTRACTABLE),
         });
+        boolean keySensitive = (attributes[0].getBoolean() ||
+                        attributes[1].getBoolean() || !attributes[2].getBoolean());
         if ((SunPKCS11.mysunpkcs11 != null) && "RSA".equals(algorithm)) {
-            if (attributes[0].getBoolean() || attributes[1].getBoolean() || (attributes[2].getBoolean() == false)) {
+            if (keySensitive) {
```

Comment:
Rather than nested `if` statement, suggest making this more like newer versions:
```java
        if (keySensitive && (SunPKCS11.mysunpkcs11 != null) && "RSA".equals(algorithm)) {
```
also lines 411-412 below:
```java
        if (keySensitive && (SunPKCS11.mysunpkcs11 != null) && "EC".equals(algorithm)) {
```

Reply:
Updated